### DISCUSSION TITLE
umi框架的请求返回问题

### DISCUSSION THREAD

**Discussion by shao-mingxuan** [umijs/umi#12374] [General]
1.import { history, request } from 'umi';
2.import request, { extend, RequestOptionsInit } from 'umi-request';
这两种 umi自带的请求，如果后端默认返回 success =false；前端是拿不到数据吗？
目前做的一些测试，请求后端返回的接口 success=false；这是在同一个接口的情况下，
1.从umi 解构的request 是拿不到数据，直接走的是catch方法，
2.但是从umi-request 解构的request就能拿到 success=false的这种情况，前端then方法能拿到
3.从umi-request 解构的request ，使用extend实例去请求，有的时候也是拿不到 success =false的这种情况
目前就是这三种情况，前提是没有配置错误处理和响应拦截。
从官方umi框架，约定好的接口返回success=false，是错误处理，但是出现上边的第二个情况，这是什么问题。

**Comment by fz6m** [comment]
See #12373
